source/extensions/http/cache/file_system_http_cache/cache_file_fixed_block.cc

```diff
@@ -0,0 +1,94 @@
+#include "source/extensions/http/cache/file_system_http_cache/cache_file_fixed_block.h"
+
+#include "source/common/common/assert.h"
+#include "source/common/common/safe_memcpy.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+namespace FileSystemHttpCache {
+
+namespace {
+// The expected first four bytes of the header - if fileId() doesn't match ExpectedFileId
+// then the file is not a cache file and should be removed from the cache.
+// Beginning of file should be "CACH".
+constexpr std::array<char, 4> ExpectedFileId = {'C', 'A', 'C', 'H'};
+
+// The expected next four bytes of the header - if cacheVersionId() doesn't match
+// ExpectedCacheVersionId then the file is from an incompatible cache version and should
+// be removed from the cache.
+// Next 4 bytes of file should be "0000".
+constexpr std::array<char, 4> ExpectedCacheVersionId = {'0', '0', '0', '0'};
+
+// Deserialize 4 bytes into a uint32_t.
+uint32_t deserializeUint32(const char* p) {
+  // chars must be cast to uint8_t first so that e.g. -1 becomes 0xff, before casting to uint32_t
+  // - otherwise -1 becomes 0xffffffff.
+  return static_cast<uint32_t>(static_cast<uint8_t>(p[0])) << 24 |
+         static_cast<uint32_t>(static_cast<uint8_t>(p[1])) << 16 |
+         static_cast<uint32_t>(static_cast<uint8_t>(p[2])) << 8 |
+         static_cast<uint32_t>(static_cast<uint8_t>(p[3]));
+}
+
+// Deserialize 8 bytes into a uint64_t.
+uint64_t deserializeUint64(const char* p) {
+  return static_cast<uint64_t>(deserializeUint32(p)) | deserializeUint32(p + 4);
```

Comment:
Hm I thought I commented on this already but I think this is wrong (need to left-shift one of those). Can this be (a) moved to common/common (where maybe we'll find similar functionality we can either de-dup or put near) and (b) unit-tested?

I don't actually see anything specific but maybe in common/network/utility.h or common/common/utility.h in class StringUtil ?

There's no particular synergy with the existing functions there but it would just be the kind of place I'd look for such functions.

Reply:
I think these could be replaced with absl::big_endian::Load[32|64] and Store[32|64]

Reply:
Thanks so much for knowing about those. 100% not matching anything I would have searched for when looking for these functions.